File: algorithms/2_medium/M/03979_maximum_valid_pair_sum/solution.c
static inline int max_int(const int a, const int b)
{
    return a > b ? a : b;
}

int maxValidPairSum(const int* nums, const int len, const int k)
{
    assert(k > 0 && k < len && nums != NULL);

    int max_lo = INT_MIN;
    int res = INT_MIN;
    const int end = len - k;
    for (int i = 0; i < end; i += 1) {
        max_lo = max_int(max_lo, nums[i]);
        res = max_int(res, max_lo + nums[i + k]);
    }

    return res;
}

File: algorithms/2_medium/M/03979_maximum_valid_pair_sum/solution.rs
impl Solution {
    pub fn max_valid_pair_sum(nums: Vec<i32>, k: i32) -> i32 {
        assert!(k > 0 && (k as usize) < nums.len());
        let k = k as usize;
        let len = nums.len();

        let mut max_lo = i32::MIN;
        let mut res = i32::MIN;
        let end = len - k;
        for i in 0..end {
            max_lo = max_lo.max(nums[i]);
            res = res.max(max_lo + nums[i + k]);
        }

        res
    }
}
